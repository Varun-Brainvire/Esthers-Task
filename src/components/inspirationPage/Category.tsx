import React from "react";
import {
  CategoryContainer,
  SingleCategoryCard,
  HeadContainer,
  TitleContainer,
  OverlayDiv,
  OverlayText,
} from "./category.styles";
import { CategoryImage } from "./category.styles";
import beautypic from "public/inspirationpageimages/beauty 1.jpg";
import fashionpic from "public/inspirationpageimages/Fashion 1.png";
import homepic from "public/inspirationpageimages/home 2 - alt 1.png";
import { Container, Row, Col } from "../index";
import { useRouter } from "next/router";

interface Category {
  label: string;
  imageSrc?: string | any;
}

interface CategoryProps {
  selectedCategory: string;
  onTabClick: (category: string) => void;
}

const Category: React.FC<CategoryProps> = ({
  selectedCategory,
  onTabClick,
}) => {
  const router = useRouter();
  const filters: Category[] = [
    { label: "All" },
    { label: "Beauty", imageSrc: beautypic },
    { label: "Home", imageSrc: homepic },
    { label: "Fashion", imageSrc: fashionpic },
  ];

  const getTitleContent = (tab: string) => {
    switch (tab) {
      case "All":
        return "Explore";
      case "Beauty":
        return "Beauty Inspiration";
      case "Home":
        return "Home Inspiration";
      case "Fashion":
        return "Fashion Inspiration";
      default:
        return "Explore";
    }
  };

  return (
    <>
      <HeadContainer>
        <Container>
          <Row>
            <Col lg={12}>
              <TitleContainer>
                {getTitleContent(selectedCategory)}
              </TitleContainer>
            </Col>
          </Row>
        </Container>
        <Container>
          <Row>
            <CategoryContainer>
              {filters.map((filter) => (
                <Col lg={3} sm={3} md={3}>
                  <SingleCategoryCard
                    key={filter.label}
                    onClick={() => {
                      onTabClick(filter.label);
                      if (filter.label === "All") {
                        router.push("/inspiration");
                      } else {
                        router.push(`/${filter.label.toLowerCase()}`);
                      }
                    }}
                    style={{
                      backgroundColor:
                        selectedCategory === "All"
                          ? "rgba(52, 85, 74, 1)"
                          : "rgba(244, 240, 236, 1)",
                    }}
                  >
                    {filter.imageSrc && (
                      <CategoryImage
                        src={filter.imageSrc}
                        alt={`${filter.label} pic`}
                        width={230}
                        height={180}
                        style={{
                          opacity:
                            selectedCategory === filter.label ? "1" : "0.6",
                        }}
                      />
                    )}
                    <OverlayDiv
                      style={{
                        backgroundColor:
                          selectedCategory === "All"
                            ? "rgba(52, 85, 74, 0.5)"
                            : "rgba(0, 0, 0, 0.0)",
                      }}
                    >
                      <OverlayText>{filter.label}</OverlayText>
                    </OverlayDiv>
                  </SingleCategoryCard>
                </Col>
              ))}
            </CategoryContainer>
          </Row>
        </Container>
      </HeadContainer>
    </>
  );
};

export default Category;
