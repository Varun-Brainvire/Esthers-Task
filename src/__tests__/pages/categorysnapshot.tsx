import { render } from "@testing-library/react";
import Category from "@/components/inspirationPage/Category";
import { useRouter } from "next/router";

jest.mock('next/router', () => ({
    __esModule: true,
     useRouter : jest.fn(),
  }));

it("renders category unchanged", () => {
  
  const mockOnTabClick = jest.fn();
  const { container } = render(<Category  selectedCategory="All" onTabClick={mockOnTabClick}/>)
  expect(container).toMatchSnapshot()
});