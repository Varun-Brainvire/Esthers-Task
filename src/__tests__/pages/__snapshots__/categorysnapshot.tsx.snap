// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders category unchanged 1`] = `
<div>
  <div
    class="sc-7ff8fb3c-0 gOXKjW"
  >
    <h1
      class="sc-7ff8fb3c-1 dEDASR"
    >
      Explore
    </h1>
    <div
      class="sc-bdfCDU sc-9af70b48-0 bCdRCv jukmPH"
      data-name="container"
    >
      <div
        class="sc-gsTDqH sc-ccc3d276-0 cyHZsm dONdvU"
        data-name="row"
      >
        <div
          class="sc-7ff8fb3c-2 eYGYWU"
        >
          <div
            class="sc-dlfmHC sc-bbf47a8d-0 ZCqIT gLpuEv"
            data-name="col-sm-3 col-lg-3"
            lg="3"
            sm="3"
          >
            <div
              class="sc-7ff8fb3c-3 kVrHye"
              style="background-color: rgb(52, 85, 74);"
            >
              <div
                class="sc-7ff8fb3c-4 bBWMxU"
                style="background-color: rgb(52, 85, 74);"
              >
                <span
                  class="sc-7ff8fb3c-5 eJdcXx"
                >
                  All
                </span>
              </div>
            </div>
          </div>
          <div
            class="sc-dlfmHC sc-bbf47a8d-0 ZCqIT gLpuEv"
            data-name="col-sm-3 col-lg-3"
            lg="3"
            sm="3"
          >
            <div
              class="sc-7ff8fb3c-3 kVrHye"
              style="background-color: rgb(244, 240, 236);"
            >
              <img
                alt="Beauty pic"
                class="sc-7ff8fb3c-6 leDqNG"
                data-nimg="1"
                decoding="async"
                height="180"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
                style="color: transparent; opacity: 0.6;"
                width="252"
              />
              <div
                class="sc-7ff8fb3c-4 bBWMxU"
                style="background-color: rgba(244, 240, 236, 0.4);"
              >
                <span
                  class="sc-7ff8fb3c-5 eJdcXx"
                >
                  Beauty
                </span>
              </div>
            </div>
          </div>
          <div
            class="sc-dlfmHC sc-bbf47a8d-0 ZCqIT gLpuEv"
            data-name="col-sm-3 col-lg-3"
            lg="3"
            sm="3"
          >
            <div
              class="sc-7ff8fb3c-3 kVrHye"
              style="background-color: rgb(244, 240, 236);"
            >
              <img
                alt="Home pic"
                class="sc-7ff8fb3c-6 leDqNG"
                data-nimg="1"
                decoding="async"
                height="180"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
                style="color: transparent; opacity: 0.6;"
                width="252"
              />
              <div
                class="sc-7ff8fb3c-4 bBWMxU"
                style="background-color: rgba(244, 240, 236, 0.4);"
              >
                <span
                  class="sc-7ff8fb3c-5 eJdcXx"
                >
                  Home
                </span>
              </div>
            </div>
          </div>
          <div
            class="sc-dlfmHC sc-bbf47a8d-0 ZCqIT gLpuEv"
            data-name="col-sm-3 col-lg-3"
            lg="3"
            sm="3"
          >
            <div
              class="sc-7ff8fb3c-3 kVrHye"
              style="background-color: rgb(244, 240, 236);"
            >
              <img
                alt="Fashion pic"
                class="sc-7ff8fb3c-6 leDqNG"
                data-nimg="1"
                decoding="async"
                height="180"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
                style="color: transparent; opacity: 0.6;"
                width="252"
              />
              <div
                class="sc-7ff8fb3c-4 bBWMxU"
                style="background-color: rgba(244, 240, 236, 0.4);"
              >
                <span
                  class="sc-7ff8fb3c-5 eJdcXx"
                >
                  Fashion
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
